rcl/test/rcl/test_publisher.cpp

```diff
@@ -290,3 +290,93 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_publisher_init_
   EXPECT_EQ(RCL_RET_BAD_ALLOC, ret) << rcl_get_error_string().str;
   rcl_reset_error();
 }
+
+TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_publisher_loan) {
+  rcl_ret_t ret;
+  rcl_publisher_t publisher = rcl_get_zero_initialized_publisher();
+  const rosidl_message_type_support_t * ts =
+    ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Strings);
+  const char * topic_name = "chatter";
+  rcl_publisher_options_t publisher_options = rcl_publisher_get_default_options();
+  ret = rcl_publisher_init(&publisher, this->node_ptr, ts, topic_name, &publisher_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_ret_t ret = rcl_publisher_fini(&publisher, this->node_ptr);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  });
+
+  test_msgs__msg__Strings msg_loaned;
+  if (rcl_publisher_can_loan_messages(&publisher)) {
+    EXPECT_EQ(
+      RCL_RET_OK, rcl_borrow_loaned_message(
+        &publisher,
+        ts,
+        reinterpret_cast<void **>(&msg_loaned)));
```

Comment:
@Blast545 hmm, I have the impression that `rcl_borrow_loaned_message()` returns a pointer to an allocated (loaned) message, in which case `msg_loaned` should be a pointer to `test_msgs__msg__Strings` and this cast shouldn't be necessary (implicit casts to `void` pointers are allowed)

Reply:
I fixed `msg_loaned` to be a pointer to `test_msgs__msg__Strings`, and actually added a pointer to that one for this part. However, I still got:
`invalid conversion from ‘test_msgs__msg__Strings***’ to ‘void**’ [-fpermissive]`
and had to left the cast it in my last commit.

Reply:
Hmm, why do you provide a pointer to your pointer to pointer? `msg_loaned_ptr` should be enough, no leading `&`